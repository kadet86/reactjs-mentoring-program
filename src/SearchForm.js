import React from 'react';
import SearchBy from './SearchBy';
import {Button} from 'primereact/button';
import {InputText} from 'primereact/inputtext';

export default function SearchForm({
    query, 
    searchBy, 
    onQueryChange, 
    onSearchByChange, 
    onSubmit,
}) {
    return (
        <form className="search-form" onSubmit={onSubmit}>
            <InputText 
                className="search-input"
                placeholder="Search..." 
                type="text" 
                value={query} 
                onChange={onQueryChange} />
            <SearchBy 
                searchBy={searchBy} 
                onChange={onSearchByChange} />
            <Button
                className="p-button-danger search-button" 
                type="submit" 
                label="Search" />
        </form>
    );
}