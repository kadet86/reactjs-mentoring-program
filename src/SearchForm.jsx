// @flow
import * as React from 'react';
import SearchBy from './SearchBy';
import styled from 'styled-components';
import Button from './Button';
import InputText from './InputText';

type Props = {
    query?: string,
    searchBy: string,
    onQueryChange: (event: SyntheticEvent<HTMLInputElement>) => void,
    onSearchByChange: (event: SyntheticEvent<HTMLElement>) => void,
    onSubmit: (event: SyntheticEvent<HTMLElement>) => void,
};

const Form = styled.form`
    margin-bottom: 20px;

    > button {
        font-size: 16px;
        float: right;
    }
`;

export default function SearchForm({
    query,
    searchBy,
    onQueryChange,
    onSearchByChange,
    onSubmit,
}: Props) {
    return (
        <Form className="search-form" onSubmit={onSubmit}>
            <InputText
                placeholder="Search..."
                type="text"
                required
                value={query || ''}
                onChange={onQueryChange}
            />
            <SearchBy searchBy={searchBy} onChange={onSearchByChange} />
            <Button type="submit" label="Search" className="button-danger" />
        </Form>
    );
}
