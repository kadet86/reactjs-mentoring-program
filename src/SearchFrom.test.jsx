import React from 'react';
import {shallow} from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
    it('has persistent default snapshot', () => {
        const component = shallow(<SearchForm />);
        expect(component.html()).toMatchSnapshot();
    });

    it('has persistent snapshot with query="green"', () => {
        const component = shallow(<SearchForm query="green"/>);
        expect(component.html()).toMatchSnapshot();
    });

    it('has persistent snapshot with searchBy="genres"', () => {
        const component = shallow(<SearchForm searchBy="genres"/>);
        expect(component.html()).toMatchSnapshot();
    });
});
