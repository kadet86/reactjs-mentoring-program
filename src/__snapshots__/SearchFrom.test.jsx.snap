// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component has persistent default snapshot 1`] = `ShallowWrapper {}`;

exports[`SearchForm component has persistent snapshot with query="green" 1`] = `ShallowWrapper {}`;

exports[`SearchForm component has persistent snapshot with searchBy="genres" 1`] = `ShallowWrapper {}`;
